starboard/nplb/player_get_audio_configuration_test.cc

```diff
@@ -0,0 +1,236 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+
+#include "starboard/nplb/player_test_fixture.h"
+#include "starboard/string.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+#if SB_API_VERSION >= 15
+
+bool operator==(const SbMediaAudioConfiguration& left,
+                const SbMediaAudioConfiguration& right) {
+  return memcmp(&left, &right, sizeof(SbMediaAudioConfiguration)) == 0;
+}
+
+namespace starboard {
+namespace nplb {
+namespace {
+
+using ::testing::ValuesIn;
+
+typedef SbPlayerTestFixture::GroupedSamples GroupedSamples;
+
+class SbPlayerGetAudioConfigurationTest
+    : public ::testing::TestWithParam<SbPlayerTestConfig> {
+ public:
+  void ReadAudioConfigurations(
+      const SbPlayerTestFixture& player_fixture,
+      std::vector<SbMediaAudioConfiguration>* configurations) {
+    // We try to read audio configuration with a too large index here, to make
+    // sure SbPlayerGetAudioConfiguration() not always return true and avoid
+    // infinite for loop below.
+    const int kInvalidIndex = 9999;
+    SbMediaAudioConfiguration configuration;
+    ASSERT_FALSE(SbPlayerGetAudioConfiguration(player_fixture.GetPlayer(),
+                                               kInvalidIndex, &configuration));
+
+    for (int index = 0;; index++) {
+      if (SbPlayerGetAudioConfiguration(player_fixture.GetPlayer(), index,
+                                        &configuration)) {
+        ASSERT_NO_FATAL_FAILURE(AssertConfigurationIsValid(configuration));
+        configurations->push_back(configuration);
+      } else {
+        return;
+      }
+    }
+  }
+
+  void AssertConfigurationIsValid(
+      const SbMediaAudioConfiguration& configuration) {
+    ASSERT_GE(configuration.connector, kSbMediaAudioConnectorUnknown);
+    ASSERT_LE(configuration.connector, kSbMediaAudioConnectorUsb);
+    ASSERT_GE(configuration.coding_type, kSbMediaAudioCodingTypeNone);
+    ASSERT_LE(configuration.coding_type, kSbMediaAudioCodingTypePcm);
```

Comment:
Let's implement it using switch(...), this allows us to list all valid values in the cases, and triggers build errors when we miss one configuration, so we won't miss this test when we add new values.